runtime/runtime.c

```diff
@@ -226,84 +224,19 @@ void* allocate_n_bytes_ptr(u32 n) {
     return get_memory_ptr_for_runtime(addr, n);
 }
 
-// FIXME: Currently the timer support stuff is disabled, pending cortex_m results
-// this provides a way to add a timeout for wasm execution, and support for that
-// TODO: Add a way for awsm to give us this value
-//WEAK unsigned int wasm_execution_timeout_ms = 0;
-//sigjmp_buf timeout_jump;
-//#define JUMPING_BACK 0xBACC
-
-// Precondition: you've already loaded timeout_jump with where we should jump after the timeout
-//void handle_sigalarm(int sig) {
-//    // TODO: We use siglongjmp which resets signal stuff, so perhaps this call is unnessesary
-//    signal(sig, SIG_IGN);
-//    siglongjmp(timeout_jump, JUMPING_BACK);
-//}
-//
-//void schedule_timeout() {
-//    signal(SIGALRM, handle_sigalarm);
-//    ualarm(wasm_execution_timeout_ms * 1000, 0);
-//}
-//
-//void cancel_timeout() {
-//    ualarm(0, 0);
-//}
-
 // If we are using runtime globals, we need to populate them
 WEAK void populate_globals() {}
 
-// Code that actually runs the wasm code
-IMPORT i32 wasmf_main(i32 a, i32 b);
-
-int runtime_main(int argc, char** argv) {
-    // Setup the linear memory and function table
+void runtime_init(){
     alloc_linear_memory();
     populate_table();
-
-    // Setup our allocation logic
-    runtime_heap_base = wasmg___heap_base;
-    printf("starting rhb %d\n", runtime_heap_base);
-    if (runtime_heap_base == 0) {
-        runtime_heap_base = memory_size;
-    }
-
-    // Setup the global values (if needed), and populate the linear memory
     switch_out_of_runtime();
     populate_globals();
     switch_into_runtime();
     populate_memory();
 
-    // In the case of a real timeout being compiled in, handle that
-//    if (wasm_execution_timeout_ms) {
-//        // Set the jumpoint to here, and save the signal mask
-//        int res = sigsetjmp(timeout_jump, 1);
-//        if (res != 0) {
-//            assert(res == JUMPING_BACK);
-//            printf("WE DECIDED TO GIVE UP\n");
-//            return -JUMPING_BACK;
-//        }
-//        schedule_timeout();
-//    }
-
-    u32 array_offset = allocate_n_bytes(argc * sizeof(i32));
-    u32* array_ptr = get_memory_ptr_void(array_offset, argc * sizeof(i32));
-    for (int i = 0; i < argc; i++) {
-        size_t str_size = strlen(argv[i]) + 1;
-        u32 str_offset = allocate_n_bytes(str_size);
-        char* str_ptr = get_memory_ptr_for_runtime(str_offset, str_size);
-        strcpy(str_ptr, argv[i]);
-        array_ptr[i] = str_offset;
+    runtime_heap_base = wasmg___heap_base;
+    if (runtime_heap_base == 0) {
+        runtime_heap_base = memory_size;
     }
```

Comment:
This seems... sketch

Reply:
Actionable feedback here?

Reply:
You're right, this was un-actionable. Just looking for a comment in the code on why we want to assume the heap base is `memory_size` but only sometimes.

Reply:
I honestly have no clue what this is supposed to do. This was in `runtime_main` before any of my work, and all I've done is move it into `runtime_init` since the `runtime_main` logic now differs between wasmception and WASI-SDK. 

What is currently in mainline:
![image](https://user-images.githubusercontent.com/8700322/126576772-e62595c8-5bc7-485e-b272-6d3872f97698.png)


It sounds like this should be a follow up issue since none of us really know why this is here. I opening the following for this: https://github.com/gwsystems/aWsm/issues/44. Since this isn't a change from master, I'm marking this as resolved.

Reply:
Ack. 


I do think it maybe should always be at the end of memory (not at the heap base), since otherwise malloc might overwrite this. Honestly this is basically just a wasmception hack since wasi does APIs in a more sane way. But since you didn't modify this code it isn't at all in scope for your work

Reply:
Thanks for the feedback! Adding your comment to the follow up issue.